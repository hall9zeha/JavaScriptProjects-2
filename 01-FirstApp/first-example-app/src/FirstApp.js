import React from "react";
import PropTypes from 'prop-types' 

var person = {
    name:"Martha",
    age:27
}
//const FirstApp = (props) // sin destructuring
const FirstApp = ({greeting ='Hola no se envió ningun props', subtitle}) =>{//contenido por defecto si la propiedad llega vacía
    return (
        // una forma más práctica de usar fragments es poniendo las etiquetas <> & </>
        <> 
        <h1>Hola Mundo</h1>
        {/* Imprimiendo saludo desde las propiedades recibidas */}
        {/* <h1> {props.greeting} </h1> */}{/*sin destructuring*/}
        <h1>{greeting}</h1>{/*con destructuring*/}
        <h2>{subtitle}</h2>
        <pre>{JSON.stringify(person)}</pre>
        <p>My first example in java-script after long time</p>
        </>
    )
}
//Para mostrar un mensaje de error cuando el tipo de propiedad de nuestra variable greeting no sea 
//del tipo especificado
FirstApp.propTypes ={
    //Para obligar a que la propiedad sea enviada al componente "isRequired"
    greeting: PropTypes.string.isRequired//, 
    //si no ponemos isRequired es opcional enviarla o no
    //otherProperty:PropTypes.number 
}
//Para tratar de la misma forma propiedades por defecto
FirstApp.defaultProps = {
    subtitle: 'Soy un subtítulo'
}

export default FirstApp;