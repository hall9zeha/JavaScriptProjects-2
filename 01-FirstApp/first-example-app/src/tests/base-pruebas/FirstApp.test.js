import FirstApp from "../../FirstApp";
import {render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react'

describe('test in FirstApp', () => { 

    test('should return a greeting "Hola developer"', () => { 

        const greetingTest = "Hola developer";

        const {getByText} = render(<FirstApp greeting = {greetingTest}/>);
        
        expect(getByText(greetingTest)).toBeInTheDocument();

     });

     //Probando capturas de html con snapshots
     test('should show a title sending by props',()=>{

        const greetingTest = "Hola developer";

        const wrapper = renderer.create(<FirstApp greeting={greetingTest}/>);

        expect(wrapper).toMatchSnapshot();
     });
     //Probando encontrar el contenido dentro de un elemento del componente
     test('should show a subtitle sending by props', ()=>{
        const greetingTest = "Hola developer";
        const subtitleTest="Esfuerzate mucho"
        //const wrapper = renderer.create(<FirstApp greeting={greetingTest} subtitle={subtitleTest}/>);
        render(<FirstApp greeting={greetingTest} subtitle={subtitleTest}/>)
        //Usando RTL ya que enzyme está descontinuado para versiones posteriores a React 17
        expect(screen.getByRole('heading',{level : 2})).toHaveTextContent(subtitleTest);//El subtítulo se encuentra en una etiqueta h2
     });

})