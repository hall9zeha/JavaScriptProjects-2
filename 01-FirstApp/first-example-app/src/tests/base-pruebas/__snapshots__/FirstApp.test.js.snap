// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in FirstApp should show a subtitle sending by props 1`] = `
Array [
  <h1>
    Hola Mundo
  </h1>,
  <h1>
    Hola developer
  </h1>,
  <h2>
    Soy un subtítulo
  </h2>,
  <pre>
    {"name":"Martha","age":27}
  </pre>,
  <p>
    My first example in java-script after long time
  </p>,
]
`;

exports[`test in FirstApp should show a title sending by props 1`] = `
Array [
  <h1>
    Hola Mundo
  </h1>,
  <h1>
    Hola developer
  </h1>,
  <h2>
    Soy un subtítulo
  </h2>,
  <pre>
    {"name":"Martha","age":27}
  </pre>,
  <p>
    My first example in java-script after long time
  </p>,
]
`;
